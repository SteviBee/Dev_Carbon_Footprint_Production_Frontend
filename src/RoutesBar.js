import React from 'react';
import { Routes, Route } from 'react-router-dom';

import Home from './Home';
import Login from './Login';
import Profile from './Profile';
import Signup from './Signup';



function RoutesBar() {

    return (
        <main>
            <Routes>
                <Route path="/" element={<Home />}/>             
                <Route path="login" element={<Login />}/>             
                <Route path="signup" element={<Signup />}/>             
                <Route path="profile" element={<Profile />}/>    
                <Route path="*" element={<p>Hmmm. I can't seem to find what you want.</p>}/>    
       
            </Routes>
        </main>
    );
  
}


export default RoutesBar;