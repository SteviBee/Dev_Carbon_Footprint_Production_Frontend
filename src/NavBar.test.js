import React from "react";
import { BrowserRouter } from "react-router-dom";
import UserContext from "./userContext"
import { render } from "@testing-library/react";
import NavBar from './NavBar'
import RoutesBar from "./RoutesBar"

it("renders without crashing", function () {
    let curUser = "billy"
    render(
        <UserContext.Provider value={{ curUser }}>
        <BrowserRouter>
          <NavBar />
          <RoutesBar />
        </BrowserRouter>
      </UserContext.Provider>
    );
  });

  it("matches snapshot", function () {
    let curUser = "billy"
    const { asFragment } = render(
       
            <UserContext.Provider value={{ curUser }}>
            <BrowserRouter>
              <NavBar />
              <RoutesBar />
            </BrowserRouter>
          </UserContext.Provider>
        )
    expect(asFragment()).toMatchSnapshot();
  });