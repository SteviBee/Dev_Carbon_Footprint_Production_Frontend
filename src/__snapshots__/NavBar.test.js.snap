// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="navbar navbar-expand-md"
    >
      <div
        class="container-fluid"
      >
        <a
          aria-current="page"
          class="navbar-brand active"
          href="/"
        >
          Dev Carbon
        </a>
        <ul
          class="ms-auto navbar-nav"
        >
          <li
            class="nav-item"
          >
            <a
              class=""
              href="/login"
            >
              Login
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class=""
              href="/signup"
            >
              Signup
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class=""
              href="/carbonform"
            >
              Carbon Form
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class=""
              href="/profile"
            >
              Profile
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class=""
              href="/login"
            >
              Logout
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
  <main>
    <h1>
      Developer Carbon Emission Evaluation App
    </h1>
    <h4>
      Enter application parameters - get out actions to reduce CO2 impact.
    </h4>
    <h3>
      Welcome back undefined
    </h3>
  </main>
</DocumentFragment>
`;
